### DISCUSSION TITLE
How does minIO manage different versions of an object?

### DISCUSSION THREAD

**Discussion by fengxueyoung** [minio/minio#19550] [Q&A] [answered]
If I enable the version controll feature, when I upload different versions of a file(or say an object), does minIO save the whole file straight or does it do something like compressing or slicing?
And if I upload a 1GB file, then I make another 9 versions of it with slightly difference(or even no difference) between each version,will it take me 10GB space?

**Comment by klauspost** [comment, marked as answer]
There is no de-duplication if that is what you are asking. 

That would make deleting older versions unreasonably expensive and also open a potential sidechannel attacks.

I wrote a blog post on that topic: https://blog.min.io/myths-about-deduplication-and-compression/

**Reply by fengxueyoung** [reply]
Thanks for your reply!
I read your blog and if I understand it correctly,even if an object has two same version(by saying "same" I mean the file or data is the same),minIO treats them as two independent object and will use transparent data compression to reduce the space cost.Is that right?

**Reply by klauspost** [reply]
> [A] minIO treats them as two independent object and [B] will use transparent data compression to reduce the space cost.Is that right?

[A] and [B] cannot be correct at the same time. [A] is correct. They are are independent.
